### PR TITLE
[net9.0] Fix loop CarouselViewHandler2

### PR DESCRIPTION
**Pull request by rmarinho**
### Description of Change

- Fix CarouselViewHandler controller2 looping setting the correct new version for the ItemsSource. 
- enable mappers for CarouselViewHandler2
- Enable UITests with new handlers2
- Fix header/footer CV2

### CODE REVIEW

**Review by Redth — DISMISSED** [review]

**Review by PureWeen — CHANGES_REQUESTED** [review]
Fix Failing Tests

File: src/Controls/src/Core/PublicAPI/net-ios/PublicAPI.Shipped.txt [hunk 1]
```diff
@@ -7590,7 +7590,6 @@ override Microsoft.Maui.Controls.ItemsView.OnBindingContextChanged() -> void
 override Microsoft.Maui.Controls.ItemsView.OnMeasure(double widthConstraint, double heightConstraint) -> Microsoft.Maui.SizeRequest
 override Microsoft.Maui.Controls.Label.OnBindingContextChanged() -> void
 override Microsoft.Maui.Controls.Layout.InvalidateMeasureOverride() -> void
-override Microsoft.Maui.Controls.Layout.Measure(double widthConstraint, double heightConstraint, Microsoft.Maui.Controls.MeasureFlags flags = Microsoft.Maui.Controls.MeasureFlags.None) -> Microsoft.Maui.SizeRequest
```
Comment (rmarinho) [inline src/Controls/src/Core/PublicAPI/net-ios/PublicAPI.Shipped.txt:7593]: this seems strange

File: src/Controls/src/Core/Handlers/Items2/iOS/StructuredItemsViewController2.cs [hunk 2]
```diff
@@ -36,194 +40,147 @@ protected override void Dispose(bool disposing)
...
-
-			if (footerView != null)
-				footerHeight = footerView.Frame.Height;
-
-			return new CGRect(CollectionView.Frame.X, CollectionView.Frame.Y, CollectionView.Frame.Width,
-				Math.Abs(CollectionView.Frame.Height - (headerHeight + footerHeight)));
-		}
-
-		public override void ViewWillLayoutSubviews()
+		protected virtual string DetermineViewReuseId(NSString elementKind)
```
Comment (rmarinho) [inline src/Controls/src/Core/Handlers/Items2/iOS/StructuredItemsViewController2.cs:49]: I m not sure about this implementation. We maybe should just  flat this to ItemsViewController

**Pull request merged**